cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -229,7 +247,15 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
             val splitIndex = method.getType.getName.lastIndexOf(".")
             val packageName = if (splitIndex != -1) method.getType.getName.substring(0, splitIndex).intern() else ""
             val className = if (splitIndex != -1) method.getType.getName.substring(splitIndex + 1).intern() else method.getType.getName.intern()
-
+            val resourceName = if (packageName.isEmpty) className else s"${packageName}.${className}"
+            val fileName = ClassLoaderFactory.classLoaderInfo.lookup(resourceName, ClassLoaderFactory.getClassLoader(classPath)).sourceFile
+            val methodName = if (method.getName.contains("$anonfun$")) {
+              method.getName.intern().split('$')(2).intern()
+            } else if (method.getName.contains("$")) {
+              method.getName.intern().split('$')(0).intern()
```

Comment:
not sure what you are trying to do here. This at least needs comments, and probably discussion